crates/ty_python_semantic/src/semantic_index.rs

```diff
@@ -186,7 +186,39 @@ pub(crate) fn global_scope(db: &dyn Db, file: File) -> ScopeId<'_> {
     FileScopeId::global().to_scope_id(db, file)
 }
 
-pub(crate) enum EagerSnapshotResult<'map, 'db> {
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash, get_size2::GetSize)]
+pub(crate) enum ScopeVisibility {
+    /// The scope is lazy (e.g. function, type alias scope).
+    Private,
+    /// The scope is eager (e.g. module, class, comprehension scope).
+    Public,
+}
```

Comment:
It might be just the documentation but why do we both `ScopeVisibility` and `ScopeLaziness`. I'm asking because the documentation suggests that `Private` and `Public` are just about lazyiness or are their other factors that influence whether a scope is public or private?

Reply:
Makes sense. We should then update the documentation because it makes it appear that the only defining factor is the eagerness (e.g. public, it's eager) when it's not?

Reply:
Ah, it seems I accidentally added the same comment to `ScopeVisibility` as I did to `ScopeLaziness`. Fixed.